### DISCUSSION TITLE
datasets/pose/tiger-pose/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#8492] [Docs]
# datasets/pose/tiger-pose/

Discover the versatile Tiger-Pose dataset, perfect for testing and debugging pose detection models. Learn how to get started with YOLOv8-pose model training.

https://docs.ultralytics.com/datasets/pose/tiger-pose/

**Comment by teengineer** [comment]
Hi, can yolov8-pose make classifications such as tiger standing, tiger lying down?

**Reply by glenn-jocher** [reply]
@teengineer hey there! 🐯 YOLOv8-pose can indeed classify different tiger poses like standing or lying down, but it primarily focuses on pose estimation rather than direct classification. You'd typically train the model to recognize various keypoints on the tiger, and based on the keypoints' positions, you could infer the pose.

For example, after training with the Tiger-Pose dataset, you could analyze the keypoints to determine the pose:
```python
from ultralytics import YOLO

# Load your trained model
model = YOLO('path/to/your/trained_model.pt')

# Run inference
results = model.predict(source='path/to/your/image.jpg')

# Analyze keypoints for pose classification
keypoints = results[0].keypoints
# Here you'd implement your logic to classify poses based on keypoints
```

This approach gives you the flexibility to define what constitutes a "standing" or "lying down" pose based on the keypoints' arrangement. Hope this helps! Let us know if you have more questions.

**Reply by teengineer** [reply]
Thank you.

**Reply by glenn-jocher** [reply]
Hey there! 🐅 Absolutely, YOLOv8-pose can be tailored to classify specific tiger poses by analyzing the keypoints' arrangement. While it's primarily designed for pose estimation, you can infer poses like standing or lying down based on the keypoints. Here's a quick example using the Tiger-Pose dataset:

```python
from ultralytics import YOLO

# Load your trained model
model = YOLO('path/to/your/trained_model.pt')

# Run inference on an image
results = model.predict(source='path/to/your/image.jpg')

# Analyze keypoints for pose classification
keypoints = results[0].keypoints
# Implement your logic here to classify poses based on keypoints
```

This way, you can classify the tiger's pose by examining the positions of detected keypoints. Hope this helps, and feel free to reach out if you have more questions! 🚀

**Comment by teengineer** [comment]
Hello, when i executed the predict code, i took a warning 'WARNING ⚠️ Environment does not support cv2.imshow() or PIL Image.show() google colab'. I worked on Google Colab. How can i see the results?

**Reply by glenn-jocher** [reply]
@teengineer hello! 😊 It seems you're facing an issue with displaying images in Google Colab due to its environment limitations. To view your results in Colab, you can use the following approach to display images directly in the notebook:

```python
from IPython.display import Image, display
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8n.pt')  # or your custom model

# Run prediction (ensure 'save' is True to save output images)
results = model.predict(source='path/to/your/image.jpg', save=True)

# Display the saved image
display(Image('runs/predict/exp/image0.jpg'))  # adjust the path based on your saved output
```

This method saves the output image and then uses IPython's display capabilities to show the image within your notebook. Hope this helps you see your results! Let us know if you have any more questions.

**Reply by teengineer** [reply]
Thank you @glenn-jocher, i forgot the 'save' parameter on predict func.

**Reply by glenn-jocher** [reply]
@teengineer No worries at all! 😊 I'm glad you found the issue with the 'save' parameter. If you have any more questions or need further assistance, feel free to ask. Happy coding! 🚀

**Reply by teengineer** [reply]
Hi, I want to show prediction result on video but it throw an warning, is there a way for showing? This warning is 'WARNING ⚠️ Environment does not support cv2.imshow() or PIL Image.show()'

**Reply by glenn-jocher** [reply]
Hey there! 😊 It looks like you're trying to display video predictions in an environment that doesn't support `cv2.imshow()` or `PIL Image.show()`, such as Google Colab. No worries, you can still view your video predictions by saving the output and then displaying it directly in the notebook. Here's a quick example on how you could modify your code to save the video predictions and then use `IPython.display` to show the video:

```python
from ultralytics import YOLO
from IPython.display import FileLink

# Load your model
model = YOLO('yolov8n.pt')  # or your custom model

# Run prediction on your video and save the output
results = model.predict(source='path/to/your/video.mp4', save=True)

# Assuming your output is saved as 'output.mp4' in the 'runs/predict/exp' folder
# Generate a link to download the video file
output_video_path = 'runs/predict/exp/output.mp4'
display(FileLink(output_video_path))
```

This code will provide a link to download the video file with the predictions. Just replace `'path/to/your/video.mp4'` and `output_video_path` with your actual video path and output path. Hope this helps you view your prediction results! Let me know if there's anything else I can assist you with.

**Reply by teengineer** [reply]
Thank you for your time.

**Reply by glenn-jocher** [reply]
Hey there! 😊 If you're encountering the 'WARNING ⚠️ Environment does not support cv2.imshow() or PIL Image.show()' message while trying to display video predictions in Google Colab, you can save the predictions to a video file and then display it directly in the notebook. Here's a quick example on how to do it:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8n.pt')  # or your custom model

# Run prediction on your video and save the output
results = model.predict(source='path/to/your/video.mp4', save=True)

# Use the 'files.download()' function from Google Colab to download the result video
from google.colab import files
files.download('path/to/saved/video.mp4')  # Adjust the path to your saved video
```

This way, you can view the video with predictions by downloading it. Hope this helps! Let me know if you have any more questions.

**Comment by PedroPiana** [comment]
In my dataset, do I necessarily need to have the Bounding Boxes?

**Reply by glenn-jocher** [reply]
For pose estimation tasks like those using the Tiger-Pose dataset, bounding boxes are not required. Instead, the focus is on keypoints that define specific parts of the object of interest—in this case, the tiger. Each keypoint is annotated with coordinates that represent significant positions, which are crucial for training pose estimation models. If you're working with a dataset for object detection, then bounding boxes would be necessary. 🐅

For more details on setting up your dataset for pose estimation, you can refer to the [Tiger-Pose dataset documentation](https://docs.ultralytics.com/datasets/pose/tiger-pose/).

**Comment by wcy7788** [comment]
What software should be used for the training of fish key points，I should mark a few key points?

**Reply by glenn-jocher** [reply]
@wcy7788 for training fish keypoints, you can use the YOLOv8-pose model. Annotate your dataset with keypoints and follow the training process similar to the Tiger-Pose example. For more details, refer to the Ultralytics documentation.

**Comment by wcy7788** [comment]
@glenn-jocherWhere should I go to Ultralytics to find the code for this?

**Reply by glenn-jocher** [reply]
You can find the code for the Tiger-Pose dataset in the `ultralytics/cfg/datasets/tiger-pose.yaml` file. For more details, please visit the [Tiger-Pose documentation](https://docs.ultralytics.com/datasets/pose/tiger-pose/).
